libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ir/IrOperationsBuilder.kt

```diff
@@ -601,8 +522,14 @@ internal class IrOperationsBuilder(
        */
       usedCoordinates.putType(first.type.rawType().name)
 
-      // Track argument usage
+      /**
+       * Track field and arguments usage.
+       * The field usage is only useful for data builders.
+       */
       fields.map { it.parentType }.distinct().forEach { parentType ->
+        if (schema.generateDataBuilders) {
+          usedCoordinates.putField(parentType, first.name)
```

Comment:
This sadly needs to stay here as we need field usage information for data builders (or we'd have to generate data builders for the complete schema which sounds like a waste).